ledger/src/blockstore_db.rs

```diff
@@ -2174,10 +2373,17 @@ where
     }
 
     pub fn put(&self, key: C::Index, value: &C::Type) -> Result<()> {
+        let is_perf_context_enabled = maybe_collect_perf_context();
```

Comment:
>  I think we could also move rocks_metrics_utils

Yes, I also think all these metric stuff should have its own file so that the code will look cleaner.  Will definitely do!

> Just the one question about whether we may want to control read/write sample rate independently.

I think we probably don't need two sample rates.  The default value should be good for both reads and writes.  In case we want 100% sample in certain debug scenarios, it also means we don't need to worry about the performance overhead so turning both read and write to 100% at the same time should be fine.

Reply:
Per a DM from @yhchiang-sol : "And I think we probably don't need to separate sample rate for reads and writes."

Fair enough, we can always add it later if we want